hive-metastore/src/main/java/org/apache/iceberg/hive/CachedClientPool.java

```diff
@@ -65,6 +65,8 @@ private synchronized void init() {
               .removalListener((key, value, cause) -> ((HiveClientPool) value).close())
               .build();
     }
+    // cleanup cache before jvm exit, avoid some clients may not be closed.
+    Runtime.getRuntime().addShutdownHook(new Thread(CachedClientPool::cleanupCache));
```

Comment:
We should close the HiveClientPool  first～

```
CachedClientPool.clientPoolCache().getIfPresent(metastoreUri);
CachedClientPool.clientPoolCache().invalidateAll();
```

Reply:
What is the downside of not closing the clients when the JVM exits? IMHO the lifetime of the client pool should be tied to the owning resources and be cleaned up through a `Closeable` interface rather than at system exit

Reply:
If a large number of tasks exit, many connections may not be closed. But as you said, this is not a serious problem.